File: forthic-py/src/tokens.py
class Token:
    pass


class StringToken(Token):
    def __init__(self, string: str):
        self.string: str = string


class CommentToken(Token):
    def __init__(self, string: str):
        self.string: str = string


class StartArrayToken(Token):
    pass


class EndArrayToken(Token):
    pass


class StartModuleToken(Token):
    def __init__(self, name: str):
        self.name: str = name


class EndModuleToken(Token):
    pass


class StartDefinitionToken(Token):
    def __init__(self, name: str):
        self.name: str = name


class EndDefinitionToken(Token):
    pass


class StartMemoToken(Token):
    def __init__(self, name: str):
        self.name: str = name


class WordToken(Token):
    def __init__(self, name: str):
        self.name: str = name


class EOSToken(Token):
    pass

File: experimental/forthic-swift/Forthic/Sources/Forthic/Tokens.swift
class Token {
}


class StringToken : Token {
    var str: String

    init(string: String) {
        self.str = string
    }
}


class CommentToken : Token {
    var str: String
    
    init(string: String) {
        self.str = string
    }
}


class StartArrayToken : Token {
}


class EndArrayToken : Token {
}


class StartModuleToken : Token {
    var name: String
    
    init(name: String) {
        self.name = name
    }
}


class EndModuleToken : Token {
}


class StartDefinitionToken : Token {
    var name: String
    
    init(name: String) {
        self.name = name
    }
}


class EndDefinitionToken : Token {
}

class WordToken : Token {
    var name: String
    
    init(name: String) {
        self.name = name
    }
}


class EOSToken: Token {
}
